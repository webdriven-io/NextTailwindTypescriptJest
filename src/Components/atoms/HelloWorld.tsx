import React from 'react';

const HelloWorld: React.FC = () => (
    <div id="HelloWorldDiv" className="bg-secondary text-primary font-serif">
        Hello World!!
    </div>
);

export default HelloWorld;
