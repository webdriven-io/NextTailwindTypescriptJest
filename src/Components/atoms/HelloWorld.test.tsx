import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import HelloWorld from './HelloWorld';

describe('Button', () => {
    it('renders a button', () => {
        render(<HelloWorld />);

        const div = document.getElementById('HelloWorldDiv');
        expect(div).toContainHTML('Hello World!!');
    });

    it('renders a button unchanged', () => {
        // eslint-disable-next-line @typescript-eslint/no-unsafe-call,@typescript-eslint/no-unsafe-member-access
        const tree: unknown = renderer.create(<HelloWorld />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
