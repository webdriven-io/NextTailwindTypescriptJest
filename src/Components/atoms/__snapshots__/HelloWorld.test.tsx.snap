// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders a button unchanged 1`] = `
<div
  className="bg-secondary text-primary font-serif"
  id="HelloWorldDiv"
>
  Hello World!!
</div>
`;
